### PR TITLE
try fix deinit abuse

### PR DESCRIPTION
**Pull request by weissi**

### CODE REVIEW

File: Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift [hunk 1]
```diff
@@ -233,7 +233,7 @@ extension NIOAsyncSequenceProducer {
             @inlinable
             deinit {
                 // We need to call finish here to resume any suspended continuation.
-                self._throwingSource.finish()
+                //self._throwingSource.finish()
```
Comment (simonjbeaumont) [inline Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift:236]: Should this make use of `self.finishOnDeinit`?

Comment (weissi) [reply Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift:236]: yes! all still WIP. hardest part will be getting all the tests to work etc :)

Comment (simonjbeaumont) [reply Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift:236]: Ack. Apologies. Didn't mean to jump the gun :)

**Comment by FranzBusch** [comment]
Closing this since we have landed this in another PR already

**Pull request closed without merging**